phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/apps/AppsController.java

```diff
@@ -85,6 +97,38 @@ public void loadApps() {
         });
 
         this.view.setOnSelectCategory(categoryDTO -> this.view.populateApps(categoryDTO.getApplications(), settingsManager));
+
+        this.view.setOnSetDefaultCategoryIcons(categoryDTOS -> {
+            // set default category icons
+            try {
+                StringBuilder cssBuilder = new StringBuilder();
+                for (CategoryDTO category : categoryDTOS) {
+                    cssBuilder.append("#" + category.getName().toLowerCase() + "Button{\n");
+                    String categoryIcon = category.getIcon();
+                    if (StringUtils.isEmpty(categoryIcon) || ! new File(new URI(categoryIcon)).exists()) {
+                        cssBuilder.append("-fx-background-image: url('/org/phoenicis/javafx/views/common/phoenicis.png');\n");
+                    } else {
+                        cssBuilder.append("-fx-background-image: url('" + categoryIcon + "');\n");
+                    }
+                    cssBuilder.append("}\n");
+                }
+                String css = cssBuilder.toString();
+                Path temp = Files.createTempFile("defaultCategoryIcons", ".css").toAbsolutePath();
+                Files.write(temp, css.getBytes());
+                String defaultCategoryIconsCss = temp.toUri().toString();
+                themeManager.setDefaultCategoryIconsCss(defaultCategoryIconsCss);
+
+                // apply current theme again to fix hierarchy
+                final String shortName = themeManager.getCurrentTheme().getShortName();
+                final String url = String.format("/org/phoenicis/javafx/themes/%s/main.css", shortName);
+                final URL style = this.getClass().getResource(url);
+                this.view.getTabPane().getScene().getStylesheets().clear();
+                this.view.getTabPane().getScene().getStylesheets().addAll(defaultCategoryIconsCss, style.toExternalForm());
```

Comment:
We can certainly discuss this. But please not in this PR as it is not really related to it. As quick reply: I agree that building the stylesheet URL here is not the most beautiful implementation.